### DISCUSSION TITLE
ePub reader ?

### DISCUSSION THREAD

**Discussion by PhilB-RLB** [TriliumNext/Trilium#2675] [Ideas] [answered]
Hello,

would it be possible to implement an ePub reader (the same way we already have a pdf reader) in Trilium Notes ?

**Comment by zadam** [comment, marked as answer]
PDF reader is provided by the electron runtime so supporting it is almost "for free". ePub would require some JS library, it's also much less generally useful, so I don't see this happening TBH.

**Reply by TheNomad11** [reply]
Trilium is already the perfect place for me for reading pdfs and storing notes about them, and then use this in writing, which I also do in Trilium because all sources are easily acessible, also by using split screen, thanks so much for this. An Epub-reader would fit in here perfectly for knowledge management and writing, I think

**Reply by ChiehSheShuai** [reply]
epub very necessary dear zadam, please backing to old zadam/trilium fork rework

**Comment by PhilB-RLB** [comment]
Ok Zadam, thank you.

**Comment by daomengRen** [comment]
Hope this feature can be implemented one day, thanks Zadam

**Reply by ChiehSheShuai** [reply]
> Hope this feature can be implemented one day, thanks Zadam

I hope so too. thanks Zadam

Even and even wider format support (cbz cbr). and remember the page you are on in the ebook, underline and add notes

One day I quit vibecoding; If I progress from there to becoming a master in algorithm design instead of becoming a real vibecoder, I would like to be a contributor to this project that I benefit from.
